File: leetcode/75.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
  let zeroPointer = 0;
  let twoPointer = nums.length - 1;
  let i = 0;
  while (i <= twoPointer) {
    if (nums[i] === 0) {
      [nums[i], nums[zeroPointer]] = [nums[zeroPointer], nums[i]];
      zeroPointer++;
      i++;
    } else if (nums[i] === 2) {
      [nums[i], nums[twoPointer]] = [nums[twoPointer], nums[i]];
      twoPointer--;
    } else {
      i++;
    }
  }
};

File: leetcode/75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        zero_pointer = 0
        two_pointer = len(nums) - 1
        i = 0
        while i <= two_pointer:
            if nums[i] == 0:
                nums[i], nums[zero_pointer] = nums[zero_pointer], nums[i]
                zero_pointer += 1
                i += 1
            elif nums[i] == 2:
                nums[i], nums[two_pointer] = nums[two_pointer], nums[i]
                two_pointer -= 1
            else:
                i += 1
                
        